### DISCUSSION TITLE
Expand to original field

### DISCUSSION THREAD

**Discussion by Aragur** [pocketbase/pocketbase#1695] [Q&A] [answered]
Hey,
I'm currently working lots of my time replacing the field with the relation ID with the actual object which is inside the expanded field.
So, I was asking myself if it's possible to do this directly by using the SDK.

For example, I don't want this response:
```json
{
  "message": "lorem ipsum",
  "user": "asdf",
  "expand":  {
    "user":  {
      "id": "asdf",
      "name": "John Doe",
      "age": 22
    }
  }
}
```

I want this directly:
```json
{
  "message": "lorem ipsum",
  "user":  {
    "id": "asdf",
    "name": "John Doe",
    "age": 22
  }
}
```

Is there any config value, that I have missed, or isn't this possible by design?

**Comment by ganigeorgiev** [comment, marked as answer]
There is no configuration for this. Please note that technically you can have more than one expanded relation, including [indirect expands](https://pocketbase.io/docs/expanding-relations/)). Relations are expanded inside `expand` for consistency so that the main model fields always have the same representation no matter whether the request client have permissions to expand or not the record.

If you are using the JS SDK and really want to replace the field key directly with the expanded result, you can make use of the `afterSend` global hook:

```js
pb.afterSend = function (response, data) {
    return Object.assign(data, {
        // extend or modify the data after each request and before resolving the Promise data...
        "additionalField": 123,
    });
};
```

**Reply by Aragur** [reply]
Thanks for your detailed answer.
But that's far too difficult, as far as I know. Because currently, this is called on each request.
So for each CORS - Request, also if I request a view of a single file the data is entirely different from if I request a List.

**Reply by ganigeorgiev** [reply]
Yes, `afterSend` is called after each request and before resolving the "final" Promise data.
There is no other global/central place where you can do this at the moment. Eventually, you can try to filter only the requests to which you want to apply the transformation by inspecting the `response` arg fields, but in general I wouldn't bother doing this and will just access the expanded result through the `expand` prop.

**Reply by Aragur** [reply]
Yes, this is now the route I am going to take :)
As it doesn't add that much more complexity to just call the `.expand.user` property.
Thanks a lot and sorry for this stupid question.
